Properly support `url` connection string in shard groups

joshfng opened the pull request:
Fixes https://github.com/thiagopradi/octopus/issues/418

thiagopradi commented:
hey @joshfng - can you provide some tests to make sure we don't break this functionality in the future? Thanks.

joshfng commented:
@thiagopradi I updated the test shard file to include a mysql2 connection string so the code performs a connection in the specs. I also added the shard name to the expected results which ensures the config was parsed correctly.

It's working for recent versions of ruby and rails, but failing for rails 4 specs. This leads me to believe there's an issue with how Octopus connects for that version. I've noticed a lot of conditionals in the codebase related to what Rails/AR version is running, one of them is how the connection is resolved, but my fiddling has been unable to make any progress.

Perhaps if you have time you can investigate further since you have a deeper understanding of the connection differences between versions? You can see that most of the build matrix is passing, it's only Rails/AR 4 that is failing https://travis-ci.org/thiagopradi/octopus/builds/306997247
